business/checkers/serviceentries/workload_entry_address_match.go

```diff
@@ -0,0 +1,77 @@
+package serviceentries
+
+import (
+	"fmt"
+
+	"istio.io/client-go/pkg/apis/networking/v1alpha3"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/kiali/kiali/models"
+)
+
+type HasMatchingWorkloadEntryAddress struct {
+	ServiceEntry    v1alpha3.ServiceEntry
+	WorkloadEntries map[string][]string
+}
+
+const MeshInternal = 1
+
+func (in HasMatchingWorkloadEntryAddress) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	if in.ServiceEntry.Spec.Location != MeshInternal {
+		return validations, true
+	}
+
+	if in.ServiceEntry.Spec.WorkloadSelector == nil {
+		return validations, true
+	}
+
+	var targetAddresses []string
+	seSelector := labels.Set(in.ServiceEntry.Spec.WorkloadSelector.Labels).AsSelector()
+
+	for labelsMap, weAddressMap := range in.WorkloadEntries {
+		workloadLabelsSet, err := labels.ConvertSelectorToLabelsMap(labelsMap)
+		if err != nil {
+			continue
+		}
+
+		if seSelector.Matches(workloadLabelsSet) {
+			targetAddresses = append(targetAddresses, weAddressMap...)
+		}
+	}
+
+	if targetAddresses == nil {
+		return validations, true
+	}
+
+	seAddresses := in.ServiceEntryAddressMap()
+
+	for _, weAddress := range targetAddresses {
+		if _, found := seAddresses[weAddress]; !found {
+			// Add validation: WorkloadEntry.Address should be part of the Service Entry Addresses list
+			validation := models.Build("serviceentries.workloadentries.addressmatch",
+				fmt.Sprint("spec/addresses"))
+			validations = append(validations, &validation)
+		}
+	}
+
+	return validations, true
```

Comment:
Yep, a checker returns false (or a validation is invalid) when there is one checker with Error Severity. In this case, we have a warning (because it doesn't break the mesh) so we can't say that this istio object is invalid.